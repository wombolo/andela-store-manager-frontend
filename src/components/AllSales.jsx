import React, { Component } from 'react';
import {connect} from "react-redux";
import { getAllSales} from "../actions/salesActions";
import {Link} from "react-router-dom";

export class AllSales extends Component {
  componentDidMount() {
    this.props.getAllSales();
  }

  render() {
    const {sales} = this.props.sales;
    const {role} = this.props.profile;

    let iteration = 0, grandTotalItems = 0, grandTotalAmount = 0;

    return (
      <div>
        <h1>My Sales</h1>

        <div className="row clearfix">
          <div className="col-10 m-0-auto m-t-5">

            <table className="dataTable table-striped" border="1" cellPadding="5">
              <thead>
              <tr>
                <td>S/N</td>
                <td>Transaction ID</td>
                <td>Item Name</td>
                <td>Item Price</td>
                <td>Item Quantity</td>
                <td>Amount</td>
                <td>Item Page</td>
                {role === 'admin'?<td> Attendant Profile </td>:null}
              </tr>
              </thead>

              <tbody>
              {sales.length > 0 ?
                sales.map(singleSale =>{
                  iteration++;

                  const thisAmountTotal = parseFloat((singleSale.price * singleSale.quantity).toFixed(2));
                  grandTotalItems += parseInt(singleSale.quantity);
                  grandTotalAmount += thisAmountTotal;

                  return(
                    <tr key={singleSale.id}>
                      <td>{iteration}</td>
                      <td><code>{Math.random().toString(36).substring(7).toUpperCase()}</code></td>
                      <td>{singleSale.title}</td>
                      <td>$ {singleSale.price}</td>
                      <td>{singleSale.quantity}</td>
                      <td>$ {thisAmountTotal}</td>
                      <td><Link to={`edit-product/${singleSale.product_id}`} className='btn btn-primary'>View</Link></td>

                      {role === 'admin'?
                        <td><Link to={`profile/${singleSale.profile_id}`} className='btn btn-dark'>View</Link></td>
                        : null
                      }

                    </tr>
                  )
                }) : <tr><td colSpan="7" className='center'>No Sales</td></tr>}
              </tbody>

              <tfoot>
              <tr>
                <td colSpan="4">Grand Total</td>
                <td>{grandTotalItems}</td>
                <td>$ {grandTotalAmount}</td>
                <td> </td>
                {role === 'admin'? <td> </td> : null}
              </tr>

              </tfoot>
            </table>

          </div>
        </div>

      </div>
    )
  }
}


export const mapStateToProps = state => ({
  sales: state.sales,
  profile: state.auth.user.profile
});


export default connect(mapStateToProps, {
  getAllSales
})(AllSales);
