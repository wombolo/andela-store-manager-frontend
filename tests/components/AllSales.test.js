import 'babel-polyfill'
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import { AllSales } from '../../src/components/AllSales.jsx';

let wrapper;

const props = {
  getAllSales: () => Promise.resolve(),
  sales:{
    sales:[
      {"id":1,"product_id":2,"profile_id":1,"title":"Things fall apart","description":"Book written by Chinua Achebe","price":"42.00","quantity":3,"cdate":"2018-10-31T21:40:17.871Z"},
      {"id":2,"product_id":2,"profile_id":1,"title":"Things fall apart","description":"Book written by Chinua Achebe","price":"42.00","quantity":3,"cdate":"2018-10-31T21:54:18.169Z"},
      {"id":3,"product_id":2,"profile_id":1,"title":"Things fall apart","description":"Book written by Chinua Achebe","price":"42.00","quantity":3,"cdate":"2018-10-31T21:57:29.032Z"},
    ]
  },
  profile: {
    id: 1,
    firstname: 'Akano',
    lastname: 'Peter',
    email: 'elkenzom1@gmail.com',
    role: 'admin',
    image: 'team-7.jpg',
  },
};

describe('AllSales Component', () => {
  it('should match snapshot', () => {
    wrapper = shallow(<AllSales {...props}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
