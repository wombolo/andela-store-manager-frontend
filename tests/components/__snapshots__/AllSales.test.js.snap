// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllSales Component should match snapshot 1`] = `
<div>
  <h1>
    My Sales
  </h1>
  <div
    className="row clearfix"
  >
    <div
      className="col-10 m-0-auto m-t-5"
    >
      <table
        border="1"
        cellPadding="5"
        className="dataTable table-striped"
      >
        <thead>
          <tr>
            <td>
              S/N
            </td>
            <td>
              Transaction ID
            </td>
            <td>
              Item Name
            </td>
            <td>
              Item Price
            </td>
            <td>
              Item Quantity
            </td>
            <td>
              Amount
            </td>
            <td>
              Item Page
            </td>
            <td>
               Attendant Profile 
            </td>
          </tr>
        </thead>
        <tbody>
          <tr
            key="1"
          >
            <td>
              1
            </td>
            <td>
              <code>
                Q94H4W
              </code>
            </td>
            <td>
              Things fall apart
            </td>
            <td>
              $ 
              42.00
            </td>
            <td>
              3
            </td>
            <td>
              $ 
              126
            </td>
            <td>
              <Link
                className="btn btn-primary"
                replace={false}
                to="edit-product/2"
              >
                View
              </Link>
            </td>
            <td>
              <Link
                className="btn btn-dark"
                replace={false}
                to="profile/1"
              >
                View
              </Link>
            </td>
          </tr>
          <tr
            key="2"
          >
            <td>
              2
            </td>
            <td>
              <code>
                GZ3D1T
              </code>
            </td>
            <td>
              Things fall apart
            </td>
            <td>
              $ 
              42.00
            </td>
            <td>
              3
            </td>
            <td>
              $ 
              126
            </td>
            <td>
              <Link
                className="btn btn-primary"
                replace={false}
                to="edit-product/2"
              >
                View
              </Link>
            </td>
            <td>
              <Link
                className="btn btn-dark"
                replace={false}
                to="profile/1"
              >
                View
              </Link>
            </td>
          </tr>
          <tr
            key="3"
          >
            <td>
              3
            </td>
            <td>
              <code>
                GCMMLB
              </code>
            </td>
            <td>
              Things fall apart
            </td>
            <td>
              $ 
              42.00
            </td>
            <td>
              3
            </td>
            <td>
              $ 
              126
            </td>
            <td>
              <Link
                className="btn btn-primary"
                replace={false}
                to="edit-product/2"
              >
                View
              </Link>
            </td>
            <td>
              <Link
                className="btn btn-dark"
                replace={false}
                to="profile/1"
              >
                View
              </Link>
            </td>
          </tr>
        </tbody>
        <tfoot>
          <tr>
            <td
              colSpan="4"
            >
              Grand Total
            </td>
            <td>
              9
            </td>
            <td>
              $ 
              378
            </td>
            <td>
               
            </td>
            <td>
               
            </td>
          </tr>
        </tfoot>
      </table>
    </div>
  </div>
</div>
`;
